tests/functional/flows/test_update_flow.py

```diff
@@ -0,0 +1,243 @@
+        "name": "The Strongest Man in the World",
+        "identity_provider": "c8abac57-560c-46c8-b386-f116ed8793d5",
+        "organization": "Personal Superheroes",
+        "status": "used",
+        "email": "artie@super.hero",
+    },
+    "monica": {
+        "username": "monica@kreb.scouts",
+        "name": "Monica Perling",
+        "identity_provider": "c8abac57-560c-46c8-b386-f116ed8793d5",
+        "organization": "Kreb Scouts",
+        "status": "used",
+        "email": "monica@kreb.scouts",
+    },
+}
+
+SPECIAL_PRINCIPALS = ["public", "all_authenticated_users"]
+
+
+class IdentityPool:
+    IDENTITY_DATA = FLOW_IDENTITIES
+
+    def __init__(self):
+        self.identities = {}
+        self.assigned_sets = {}
+
+    def assign(self, set_name, principal_set):
+        """
+        Assign a list of identities for the provided principal set, updating the
+        stored identities dict with any new identities that are created so they can
+        be reused.
+
+        set_name is the name of the key to store the assigned identities under
+        principal_set is a list of principals in the request
+        """
+        self.assigned_sets[set_name] = identity_set = []
+        # Randomize the names for each principal set
+        available_identities = list(self.IDENTITY_DATA.keys())
+        shuffle(available_identities)
+
+        # Iterate over the principals in the request
+        for index, principal in enumerate(principal_set):
+            if principal in self.identities:
+                # Use the existing identity if it's already been assigned
+                identity_set.append(self.identities[principal])
+                continue
+
+            if principal not in SPECIAL_PRINCIPALS:
+                # Attempt to assign distinct identities to each principal
+                identity = self.create_identity(
+                    principal.split(":")[-1],
+                    available_identities[index % len(available_identities)],
+                )
+                identity_set.append(identity)
+                self.identities[principal] = identity
+
+        return identity_set
+
+    def get_assigned_usernames(self, set_name):
+        """
+        Return a list of usernames for the provided set_name.
+        """
+        return [identity["username"] for identity in self.assigned_sets[set_name]]
+
+    @classmethod
+    def create_identity(cls, id, name):
+        """
+        Return an identity dict using the provided id for the user corresponding
+        to the provided name.
+        """
+        identity = cls.IDENTITY_DATA[name].copy()
+        identity["id"] = id
+        return identity
+
+
+def value_for_field_from_output(name, output):
+    """
+    Return the value for a specified field from the output of a command.
+    """
+    match = re.search(rf"^{name}:[^\S\n\r]+(?P<value>.*)$", output, flags=re.M)
+    assert match is not None
+    return match.group("value")
```

Comment:
At a glance this looks the same as the helpers in `test_create_flow`, in which case maybe we should move to a shared module?

Reply:
You are correct! I thought about doing that, but now I'll actually do it!

Reply:
Done.